### PR TITLE
Reviewer - main panel

### PR DESCRIPTION
**Pull request by webdevelopland**
What's new:
- Each time, when diff is changed, `modified_timestamp` is refreshed too.
- fix for Incoming diffs
- Additional column `workspace` is added to main panel

### CODE REVIEW

File: tools/reviewer/web/src/app/dashboard/diffs/diffs.component.ts [hunk 1]
```diff
@@ -104,7 +105,7 @@ export class DiffsComponent implements OnInit {
           } else if (needAttentionOfList.includes(userEmail)) {
             // Need attention of user
             this.diffGroups[DiffGroups.NeedAttention].push(diff);
-            this.diffGroups[DiffGroups.Outgoing].push(diff);
+            this.diffGroups[DiffGroups.Incoming].push(diff);
```
Comment (oferb) [inline tools/reviewer/web/src/app/dashboard/diffs/diffs.component.ts]: If should be in `Incoming` if `userEmail` is in reviewers, whether it's in `needsAttention` or not.

File: tools/reviewer/web/src/app/shared/services/firebase.service.ts [hunk 2]
```diff
@@ -45,6 +45,10 @@ export class FirebaseService {
   }
 
   updateDiff(diff: Diff): Observable<void> {
+    // Milliseconds since Jan 01 1970. (UTC)
```
Comment (oferb) [inline tools/reviewer/web/src/app/shared/services/firebase.service.ts]: This is commonly called `Unix timestamp`.

Comment (webdevelopland) [reply tools/reviewer/web/src/app/shared/services/firebase.service.ts]: Yeah, I know. I wanted to emphasize that it's unix timestamp in milliseconds.

Comment (oferb) [reply tools/reviewer/web/src/app/shared/services/firebase.service.ts]: So maybe rephease to `Unix timestamp in milliseconds`?

Comment (webdevelopland) [reply tools/reviewer/web/src/app/shared/services/firebase.service.ts]: Ok

File: tools/reviewer/web/src/app/dashboard/diffs/diffs.component.ts [hunk 3]
```diff
@@ -53,75 +54,81 @@ export class DiffsComponent implements OnInit {
...
+          switch (diff.getStatus()) {
+            case Diff.Status.UNDER_REVIEW:
+            case Diff.Status.NEEDS_MORE_WORK:
+            case Diff.Status.ACCEPTED:
+            case Diff.Status.SUBMITTING:
+            case Diff.Status.REVERTING:
+              // Outgoing Review
+              this.diffGroups[DiffGroups.Outgoing].push(diff);
+              break;
+            case Diff.Status.SUBMITTED:
```
Comment (oferb) [inline tools/reviewer/web/src/app/dashboard/diffs/diffs.component.ts]: Let's add REVERTED here, together with SUBMITTED.
Reason: REVERTED is a specific case of SUBMITTED.

File: tools/reviewer/web/src/app/shared/services/firebase.service.ts [hunk 4]
```diff
@@ -29,7 +29,7 @@ export class FirebaseService {
       .map(actions => {
         return actions.map(action => {
           const firebaseElement = action.payload.doc.data() as FirebaseElement;
-          return this.converFirebaseElementToDiff(firebaseElement);
+          return this.convertFirebaseElementToDiff(firebaseElement);
```
Comment (oferb) [inline tools/reviewer/web/src/app/shared/services/firebase.service.ts]: Thanks!

File: tools/reviewer/web/tslint.json [hunk 5]
```diff
@@ -8,7 +8,7 @@
     "no-var-keyword": true,
     "quotemark": [true, "single", "avoid-escape"],
     "semicolon": [true, "always", "ignore-bound-class-methods"],
-    "max-line-length": [true, 79],
+    "max-line-length": [true, 100],
```
Comment (oferb) [inline tools/reviewer/web/tslint.json]: :champagne:

**Comment by oferb** [comment]
Let's revert the time change on the diff page, so it shows the full date + time.

File: tools/reviewer/web/tslint.json (hunk 5 above)
Comment (webdevelopland) [reply tools/reviewer/web/tslint.json]: :smiley:

File: tools/reviewer/web/src/app/shared/services/firebase.service.ts (hunk 4 above)
Comment (webdevelopland) [reply tools/reviewer/web/src/app/shared/services/firebase.service.ts]: :)

**Pull request merged**